pipelines/utils/execute_dbt_model/utils.py

```diff
@@ -36,6 +40,74 @@ def get_dbt_client(
     )
 
 
+def update_profiles_for_env_credentials(dbt_repository_path: str) -> None:
+    """
+    Update the DBT profiles.yml to use environment variables for BigQuery authentication.
+
+    This function modifies the profiles.yml file to use environment variables
+    directly for authentication instead of referencing a keyfile on disk.
+
+    Args:
+        dbt_repository_path (str): Path to the DBT repository
+
+    Raises:
+        Exception: If profiles.yml cannot be found or updated
+    """
+
+    possible_profile_locations = [
+        os.path.join(dbt_repository_path, "profiles.yml"),
+        os.path.join(dbt_repository_path, ".dbt", "profiles.yml"),
+        os.path.expanduser("~/.dbt/profiles.yml"),
+    ]
+
+    profiles_path = None
+    for path in possible_profile_locations:
+        if os.path.exists(path):
+            profiles_path = path
+            break
+
+    if not profiles_path:
+        raise Exception(
+            "profiles.yml not found in any of the expected locations"
+        )
+
+    log(f"Found profiles.yml at {profiles_path}", level="info")
+
+    with open(profiles_path, "r") as f:
+        profiles = yaml.safe_load(f)
+
+    modified = False
+    for profile_name, profile in profiles.items():
```

Comment:
achei essa parte ligeiramente confusa, pode ser que eu esteja falando besteira. mas me parece que se o arquivo de profiles não estiver configurado para usar um keyfile ele vai dar o aviso "No BigQuery profiles found that use keyfile authentication" . Me parece que seria bom adicionar algo que verifica se o profile ta configurado para `environment-based authentication`, faz sentido?

Reply:
Pelo que entendi ele remover a propriedade `keyfile`, que o caminho para o arquivo da conta de serviço, e depois adiciona `method: oauth`.